py-polars/tests/unit/operations/test_join.py

```diff
@@ -1243,6 +1243,20 @@ def test_join_preserve_order_left() -> None:
         5,
     ]
 
+    right_left = left.join(right, on="a", how="right", maintain_order="left").collect()
```

Comment:
It feels a bit odd to have tests for right joins in the `test_join_preserve_order_left` function, but oh well.
IMO the fact that the right join is implemented in terms of the left join is just a (clever) implementation detail.
Leaning on that was what led to this bug in the first place 😁